### PR TITLE
cdc, mvcc: always read old value when cdc requires

### LINKED ISSUE #9981: CDC endpoint thread CPU usage 100% with insert 800ops
### What version of TiKV are you using?

tikv v4.0.6
ticdc v4.0.9

### Steps to reproduce

Creates a changefeed with old value, and executes 800ops insertion in upstream TiDB. 

### What did you expect?

CPU usage should be no more than 50%.

### LINKED ISSUE #9996: TiKV OOM due to CDC endpoint CPU 100% under 20k insert op/s
## Bug Report

image
image
image
image
image

From the above graph, OOM is caused by endpoint CPU usage is high, TiKV buffers too many scan events in memory.

### What version of TiKV are you using?

TiKV 5.0.0
TiCDC 5.0.0

### What operating system and CPU are you using?

TiKV: 8 CPU, 64GB memory

### What did you expect?

No TiKV OOM

### What did happened?

TiKV OOM

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number:

Close #9996
Close #9981

Reading old value costs too much CPU for CDC endpoint thread, leading to OOM. 

### What is changed and how it works?

Always read old value in txn layer when cdc requires old value.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

Fix CDC OOM issue caused by reading old values.

### CODE REVIEW

**Comment by 5kbpers** [comment]
What about performance?

**Comment by overvenus** [comment]
I did a benchmark in release-4.0 branch, the results can be applied to 5.0 and master too.

<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-087e7797-7fff-c647-1e99-9358233703ef"><h2 dir="ltr" style="line-height:1.38;margin-top:18pt;margin-bottom:6pt;"><span style="font-size:16pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Performance Results</span></h2><br><div dir="ltr" style="margin-left:0pt;" align="left">
&nbsp; | sysbench prepare(batch insert/qps) | sysbench oltp write_only run(insert, delete, update/qps)
-- | -- | --
1 tikv/v4.0.12 | 40 | 4k
1 tikv/cdc without old value | 35 | 3.8k
1 tikv/cdc with old value | 35 | 3.8k
4 tikv/v4.0.12 | 47 | 3.9k
4 tikv/cdc without old value | 47 | 3.8k
4 tikv/cdc with old value | 47 | 3.8k

</div></b>

1 tikv: one tikv node cluster
4 tikv: four tikv nodes cluster

Sysbench

```sh
sysbench --db-driver=mysql --mysql-host=172.16.4.55 --mysql-port=47904 \
  --mysql-user=root --time=1500 --report-interval=10 --threads=64 \
  --mysql-db=test --tables=10 --table-size=10000000 oltp_write_only prepare
```

File: src/storage/txn/actions/prewrite.rs [hunk 1]
```diff
@@ -85,18 +85,25 @@ pub fn prewrite<S: Snapshot>(
...
-            reader.get_old_value(&mutation.key, w)?
+            if mutation.txn_props.is_pessimistic() {
+                // Pessimistic transaction always skip constraint check in
+                // "prewrite" stage, as it checks constraint in
+                // "acquire pessimistic lock" stage.
+                OldValue::Unspecified
+            } else {
+                // In optimistic transaction, caller ensures that there is no
+                // previous write for the mutation, so there is no old value.
+                OldValue::None
```
Comment (MyonKeminta) [inline src/storage/txn/actions/prewrite.rs:96]: Is this correct? 😕

**Review by MyonKeminta — APPROVED** [review]

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**